File: DONGUK/프로그래머스_WEEK 3/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    
    d = {}
    
    for human in completion:
        if human not in d:
            d[human] = 1
        
        else:
            d[human] += 1
    
    
    for human in participant:
        if human not in d:
            answer = human
            break
            
        if d[human] > 0:
            d[human] -= 1
        
        else:
            answer = human
            break
    
    return answer

File: JINSEOP/프로그래머스_WEEK3/완주하지 못한 선수.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

string solution(vector<string> participant, vector<string> completion) {
    map<string, int> info;
    string answer = "";
    
    for(int comp = 0; comp<completion.size(); comp++) info[completion[comp]]++;
    for(int part = 0; part<participant.size(); part++) info[participant[part]]++;
    for(auto i : info){
        if(i.second % 2 == 1) answer = i.first;
    }
    
    return answer;
}
